src/content/doc-sdk-dotnet/data-types.mdx

```diff
@@ -300,5 +316,80 @@ var rid = new StringRecordId("person:john");
 
 // Alternatively, a StringRecordId can be inferred explicitly from a string
 var rid = (StringRecordId)"person:john";
-await client.Select((StringRecordId)"person:john");
-```
\ No newline at end of file
+await client.Select<Person>((StringRecordId)"person:john");
+```
+
+### Working with `RecordIdOfString`
+
+For string-based identifiers, you can also use the specialized type RecordIdOfString:
+
+```csharp title="Using RecordIdOfString"
+var rid = new RecordIdOfString("person", "john");
+// or
+Console.WriteLine(rid.Table); // "person"
+Console.WriteLine(rid.Id);    // "john"
+```
+
+### Working with `RecordIdOf<T>`
+
+For complex or structured identifiers, use the generic type RecordIdOf\<T\>:
+
+```csharp title="Using RecordIdOf<T>"
+public class CityId
+{
+    public string City { get; set; } = string.Empty;
+}
+
+var rid = new RecordIdOf<CityId>("city", new CityId { City = "London" });
+```
+
+This enables strongly-typed IDs that map directly to your domain objects.
+
+### Inheriting from Record
+
+If your model class inherits from Record, it will automatically include an Id property of type `RecordId`.
+
+```csharp title="Inheriting from Record"
+public class Person : Record
+{
+    public string Name { get; set; } = string.Empty;
+}
+
+// Example usage
+var person = new Person { Name = "Alice" };
+Console.WriteLine(person.Id); // RecordId ("person", "…")
+```
+
+### Using Data Annotations
```

Comment:
todo: this is generally more of an edge case and that may need more explanation. I would likely keep this page short and also see a new `Serialization` section with a note here that links to this new section.